### PR TITLE
docs: @fuel-wallet/react

### PR DESCRIPTION
**Pull request by cold-briu**
Basic documentation for the "@fuel-wallet/react" package. 

Missing code examples integration to keep it simple and catch up with the release of fuel-wallet 0.13

- Includes a "React Hooks" section in the reference providing simple descriptions, code snippets and type return definitions.
- Includes some code snippets and explanations for the "Getting started", "connecting" and "accounts" section.

### CODE REVIEW

File: packages/docs/docs/dev/getting-started.mdx [hunk 1]
```diff
@@ -27,17 +27,17 @@ To use it, you must also install the [`fuels`](https://www.npmjs.com/package/fue
...
 ### Usage
 
-To use the SDK in your TypeScript project, add the following line to your tsconfig.json file:
+To use the SDK and the React Hooks in your TypeScript project, add the following line to your tsconfig.json file:
 
 ```json
 {
   "compilerOptions": {
-    "types": ["@fuel-wallet/sdk"]
+    "types": ["@fuel-wallet/sdk", "@fuel-wallet/react"]
```
Comment (luizstacio) [inline packages/docs/docs/dev/getting-started.mdx:40]: This should not be required when using `@fuel-wallet/react` the types are auto-imported.

File: packages/docs/docs/dev/getting-started.mdx [hunk 2]
```diff
@@ -46,6 +46,7 @@ Alternatively, you can use a [TypeScript reference](https://www.typescriptlang.o
 
 ```ts
 /// <reference types="@fuel-wallet/sdk" />
+/// <reference types="@fuel-wallet/react" />
```
Comment (luizstacio) [inline packages/docs/docs/dev/getting-started.mdx:49]: Also this is not required when suing with react.

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/docs/docs/dev/accounts.mdx [hunk 3]
```diff
@@ -40,3 +40,14 @@ To listen to current account events, you can use the `fuel.events.currentAccount
   commentBlock="eventCurrentAccount"
 />
 <Examples.Events.CurrentAccount />
+
+### With React
+
+You can keep track of the current account being used just by invoking the `useAccount` hook as shown below:
```
Comment (luizstacio) [inline packages/docs/docs/dev/accounts.mdx:46]:
```suggestion
You can keep track of the current account when using React. Using the `useAccount` hook:
```

File: packages/docs/docs/dev/getting-started.mdx [hunk 4]
```diff
@@ -69,20 +69,46 @@ const fuel = new Fuel();
...
+      <App />
+    </FuelProvider>
+  </React.StrictMode>
+);
+```
+
+Here is how you can use the `useFuel` hook in a component to access an instance of `Fuel`:
+
+```ts
+import { Fuel } from '@fuel-wallet/react';
```
Comment (luizstacio) [inline packages/docs/docs/dev/getting-started.mdx:110]:
```suggestion
import { useFuel } from '@fuel-wallet/react';
```

File: packages/docs/docs/dev/reference.mdx [hunk 5]
```diff
@@ -271,3 +271,163 @@ This event is triggered when the Connectors list is updated. The callback functi
   file="../../examples/Connectors.tsx"
   commentBlock="eventConnectors"
 />
+
+## React Hooks
```
Comment (luizstacio) [inline packages/docs/docs/dev/reference.mdx:275]: I think it's better to create a new file, react hooks, to not mix the API of the fuel SDK with the hooks lib.

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/docs/docs/dev/reference.mdx (hunk 5 above)
Comment (cold-briu) [reply packages/docs/docs/dev/reference.mdx:275]: I agree on this. 

cc @sarahschwartz

File: packages/docs/docs/dev/connecting.mdx [hunk 6]
```diff
@@ -39,14 +39,28 @@ listen for changes in the connection.
...
-
-<CodeImport file="../../src/hooks/useIsConnected.tsx" />
-
-Here is how you can use the `useIsConnected` hook in a component:
-
-<CodeImport
-  file="../../examples/events/Accounts.tsx"
-  commentBlock="useisConnected"
-/>
+In a React app, you can use the `useIsConnected` hook below to check if the user's wallet is IsConnected
```
Comment (sarahschwartz) [inline packages/docs/docs/dev/connecting.mdx:42]:
```suggestion
In a React app, you can use the `useIsConnected` hook below to check if the user's wallet is connected.
```

File: packages/docs/docs/dev/hooks-reference.mdx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+  assetId: '0x000000000...',
+});
+
+console.log(balance);
+// 1000 (example balance)
+```
+
+[See the source file](https://github.com/FuelLabs/fuels-wallet/blob/master/packages/react/src/hooks/useBalance.tsx)
+
+## useChain
```
Comment (sarahschwartz) [inline packages/docs/docs/dev/hooks-reference.mdx:36]: We should show this in the networks page too.

Comment (cold-briu) [reply packages/docs/docs/dev/hooks-reference.mdx:36]: Thanks for the review!
I've added examples for networks, accounts, and signing;

File: packages/docs/docs/dev/assets.mdx [hunk 8]
```diff
@@ -38,3 +38,20 @@ To listen to asset events, you can use the `fuel.events.assets` event.
   commentBlock="eventAssets"
 />
 <Examples.Events.Assets />
+
+## With React
+
+In a react app you can use the `useAccount` hook as shown below to check the account balance for an specific asset.
```
Comment (sarahschwartz) [inline packages/docs/docs/dev/assets.mdx:44]:
```suggestion
In a React app you can use the `useAccount` hook as shown below to check the account balance for a specific asset.
```

File: packages/docs/docs/dev/accounts.mdx [hunk 9]
```diff
@@ -40,3 +40,15 @@ To listen to current account events, you can use the `fuel.events.currentAccount
   commentBlock="eventCurrentAccount"
 />
 <Examples.Events.CurrentAccount />
+
+## With React
+
+You can keep track of the current account when using React. Using the `useAccount` hook as shown below:
```
Comment (sarahschwartz) [inline packages/docs/docs/dev/accounts.mdx:46]:
```suggestion
You can keep track of the current account when using React using the `useAccount` hook as shown below:
```

File: packages/docs/docs/dev/connecting.mdx [hunk 10]
```diff
@@ -37,16 +37,30 @@ listen for changes in the connection.
...
-  file="../../examples/events/Accounts.tsx"
-  commentBlock="useisConnected"
-/>
+```ts
+import { useIsConnected } from '@fuel-wallet/react';
+// ...
+const { isConnected } = useIsConnected();
+```
+
+You can import the `useConnect` hook to access the 'connect' function and request a connection.
```
Comment (sarahschwartz) [inline packages/docs/docs/dev/connecting.mdx:50]:
```suggestion
You can import the `useConnect` hook to access the `connect` function and request a connection.
```

File: packages/docs/docs/dev/hooks-reference.mdx [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+
+```tsx
+const { transaction } = useTransaction({ txId: 'fuel1r20zhd...' });
+```
+
+[See the source file](https://github.com/FuelLabs/fuels-wallet/blob/master/packages/react/src/hooks/useTransaction.tsx)
+
+## useTransactionReceipts
+
+Retrieves transaction receipts `<TransactionResponse.Receipts>` associated with a specific transaction ID using the useFuel hook.
```
Comment (sarahschwartz) [inline packages/docs/docs/dev/hooks-reference.mdx:136]:
```suggestion
Retrieves transaction receipts `<TransactionResponse.Receipts>` associated with a specific transaction ID using the `useFuel` hook.
```

File: packages/docs/docs/dev/networks.mdx [hunk 12]
```diff
@@ -30,3 +30,15 @@ To listen to network events, you can use the `fuel.events.network` event.
 Switch the network in your wallet to see the event triggered.
 
 <Examples.Events.Network />
+
+## With React
+
+In a react app you can use the `useChain` hook as shown below to get information about the current chain.
```
Comment (sarahschwartz) [inline packages/docs/docs/dev/networks.mdx:36]:
```suggestion
In a React app you can use the `useChain` hook as shown below to get information about the current chain.
```

File: packages/docs/docs/dev/signing.mdx [hunk 13]
```diff
@@ -10,3 +10,18 @@ With access to the user address and the connection authorized, you can request t
 <CodeImport file="../../examples/SignMessage.tsx" />
 
 <Examples.SignMessage />
+
+## With React
+
+In a react app, once the connection is established, you can use the `useWallet` hook to get a wallet instance and sign the transaction.
```
Comment (sarahschwartz) [inline packages/docs/docs/dev/signing.mdx:16]:
```suggestion
In a React app, once the connection is established, you can use the `useWallet` hook to get a wallet instance and sign the transaction.
```

**Review by sarahschwartz — COMMENTED** [review]
It's looking really good! I just found a few more small nits.

File: packages/docs/docs/dev/hooks-reference.mdx [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+
+## useWindowFuel
+
+Provides access to the `Fuel Wallet` instance available in the global window object. It sets up an event listener for the `FuelLoaded` event.
+
+```tsx
+const fuel = useWindowFuel();
+```
+
+[See the source file](https://github.com/FuelLabs/fuels-wallet/blob/master/packages/react/src/hooks/useWindowFuel.tsx)
```
Comment (luizstacio) [inline packages/docs/docs/dev/hooks-reference.mdx:164]: We removed the `useWindowFuel` hook. Can we remove this?

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/react/README.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+    "types": ["@fuel-wallet/react"]
+  }
+}
+```
+
+Alternatively, you can use a [TypeScript reference](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html) directive in any TypeScript file:
+
+```ts
+/// <reference types="@fuel-wallet/react" />
+```
```
Comment (luizstacio) [inline packages/react/README.md:32]: This package works a bit different we can remove this from here

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Review by luizstacio — APPROVED** [review]

**Pull request merged**